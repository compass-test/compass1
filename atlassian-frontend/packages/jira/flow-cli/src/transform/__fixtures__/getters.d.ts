export default class TestGetters {
  constructor();
  get dom(): HTMLElement | undefined;
  get performanceOptions(): {
    enabled: boolean;
    samplingRate: number;
    slowThreshold: number;
  };
}
