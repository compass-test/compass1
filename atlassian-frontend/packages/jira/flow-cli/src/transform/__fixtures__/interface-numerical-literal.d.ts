export interface ListItemArray
  extends Array<
    Paragraph | OrderedList | BulletList | MediaSingle | CodeBlock
  > {
  0: Paragraph | MediaSingle | CodeBlock;
}
