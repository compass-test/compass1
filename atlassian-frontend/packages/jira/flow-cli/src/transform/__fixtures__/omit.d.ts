export interface A {
  foo: string;
  bar: number;
  baz: string;
}

export interface B {
  bfoo: string;
  bbar: number;
  bbaz: number;
}

export interface C extends A, B {
  cfoo: string;
}

export interface D extends Array<A>, B {
  cfoo: string;
}

export interface OptionalAndComplexSignatures {
  status: CardStatus;
  isCardVisible: boolean;
  previewOrientation: number;
  isPlayingFile: boolean;
  mediaViewerSelectedItem?: Identifier;
  metadata?: FileDetails;
  dataURI?: string;
  progress?: number;
  error?: Error;
}

export type OneKeyOmit = Omit<OptionalAndComplexSignatures, 'error'>;
export type OmitWithKeyOfOperator = Omit<OptionalAndComplexSignatures, keyof C>;
export type OmitWithTypeReference = Omit<
  OptionalAndComplexSignatures,
  Pick<A & B, 'bar' | 'bfoo'>
>;
export type MultipleKeyOmit = Omit<
  OptionalAndComplexSignatures,
  'error' | 'isCardVisible'
>;

export type OmitWithInheritInterface = Omit<C, 'cfoo'>;
export type OmitWithInheritInterfaceAndComplexTypeInHeritage = Omit<D, 'cfoo'>;
