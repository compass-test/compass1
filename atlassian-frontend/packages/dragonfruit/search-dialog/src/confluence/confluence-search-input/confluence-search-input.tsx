import React, { FunctionComponent } from 'react';
import {
  ProductSearchInput,
  BaseInputProps,
} from '../../common/product-search-input';
import { injectIntl, InjectedIntlProps } from 'react-intl';
import { messages } from '../../messages';
import { usePrimarySiteConfluenceAdvancedSearchUrlFactory } from '../confluence-utils/confluence-url-utils';
import { AdvancedSearchLinkSubjectId } from '../../common/analytics';
import { ContextualAnalyticsData, SCREEN } from '@atlassian/analytics-bridge';

interface AdditionalProps {
  debounceTime: number;
  forwardRef: React.Ref<HTMLInputElement>;
  isLoading: boolean;
  onNavigate: (href: string, event: React.MouseEvent | KeyboardEvent) => void;
  onOpen: () => void;
}

export type Props = BaseInputProps & AdditionalProps;

const ConfluenceSearchInputBase: FunctionComponent<
  Props & InjectedIntlProps
> = ({ intl, ...rest }) => {
  const confluenceAdvancedSearchUrlFactory = usePrimarySiteConfluenceAdvancedSearchUrlFactory();
  const advancedSearchUrl = confluenceAdvancedSearchUrlFactory(rest.value);
  return (
    <ContextualAnalyticsData
      sourceName="searchDialogPreQuery"
      sourceType={SCREEN}
    >
      <ProductSearchInput
        {...rest}
        advancedSearchURL={advancedSearchUrl}
        actionSubjectId={AdvancedSearchLinkSubjectId.NO_RESULTS}
        expandedPlaceholder={intl.formatMessage(
          messages.compass_search_input_expanded_placeholder,
        )}
        collapsedPlaceholder={intl.formatMessage(
          messages.common_search_input_collapsed_placeholder,
        )}
      />
    </ContextualAnalyticsData>
  );
};

export const ConfluenceSearchInput = injectIntl(ConfluenceSearchInputBase);

export default ConfluenceSearchInput;
