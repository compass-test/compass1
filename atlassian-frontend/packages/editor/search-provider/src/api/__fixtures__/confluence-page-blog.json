{
  "scopes": [
    {
      "id": "confluence.page,blogpost",
      "results": [
        {
          "content": {
            "id": "1538426521",
            "type": "page"
          },
          "container": {
            "title": "Editor Media",
            "displayUrl": "/spaces/EM"
          },
          "title": "Editor Media Planning - 2020-07-07",
          "url": "/spaces/EM/pages/1538426521/Editor+Media+Planning+-+2020-07-07",
          "baseUrl": "https://product-fabric.atlassian.net/wiki",
          "iconCssClass": "aui-icon aui-icon-small aui-iconfont-page-default",
          "entityType": "content",
          "lastModified": "2020-07-07T03:24:39+00:00",
          "excerpt": ""
        },
        {
          "content": {
            "id": "873989347",
            "type": "page"
          },
          "container": {
            "title": "Media Platform Team",
            "displayUrl": "/spaces/FIL"
          },
          "title": "Weekly Media Triad Meeting",
          "url": "/spaces/FIL/pages/873989347/Weekly+Media+Triad+Meeting",
          "baseUrl": "https://product-fabric.atlassian.net/wiki",
          "iconCssClass": "aui-icon aui-icon-small aui-iconfont-page-default",
          "entityType": "content",
          "lastModified": "2020-07-05T23:30:42+00:00",
          "excerpt": ""
        },
        {
          "content": {
            "id": "1356074858",
            "type": "page"
          },
          "container": {
            "title": "Editor Media",
            "displayUrl": "/spaces/EM"
          },
          "title": "Media test Media single",
          "url": "/spaces/EM/pages/1356074858/Media+test+Media+single",
          "baseUrl": "https://product-fabric.atlassian.net/wiki",
          "iconCssClass": "aui-icon aui-icon-small aui-iconfont-page-default",
          "entityType": "content",
          "lastModified": "2020-05-05T06:06:28+00:00",
          "excerpt": ""
        },
        {
          "content": {
            "id": "1385631549",
            "type": "page"
          },
          "container": {
            "title": "Editor Media",
            "displayUrl": "/spaces/EM"
          },
          "title": "Editor Media OKRs",
          "url": "/spaces/EM/pages/1385631549/Editor+Media+OKRs",
          "baseUrl": "https://product-fabric.atlassian.net/wiki",
          "iconCssClass": "aui-icon aui-icon-small aui-iconfont-page-default",
          "entityType": "content",
          "lastModified": "2020-06-09T02:21:27+00:00",
          "excerpt": ""
        },
        {
          "content": {
            "id": "998082697",
            "type": "superpage"
          },
          "container": {
            "title": "Editor",
            "displayUrl": "/spaces/E"
          },
          "title": "Media linking",
          "url": "/spaces/E/pages/998082697/Media+linking",
          "baseUrl": "https://product-fabric.atlassian.net/wiki",
          "iconCssClass": "aui-icon aui-icon-small aui-iconfont-page-default",
          "entityType": "content",
          "lastModified": "2020-04-15T03:00:05+00:00",
          "excerpt": ""
        },
        {
          "content": {
            "id": "998082697",
            "type": "blogpost"
          },
          "container": {
            "title": "Editor",
            "displayUrl": "/spaces/E"
          },
          "title": "Media linking",
          "url": "/spaces/E/pages/998082697/Media+linking",
          "baseUrl": "https://product-fabric.atlassian.net/wiki",
          "iconCssClass": "aui-icon aui-icon-small aui-iconfont-page-default",
          "entityType": "content",
          "lastModified": "2020-04-15T03:00:05+00:00",
          "excerpt": ""
        }
      ],
      "size": 4871,
      "abTest": {
        "abTestId": "default",
        "controlId": "default",
        "experimentId": "content-searcher"
      }
    }
  ]
}
