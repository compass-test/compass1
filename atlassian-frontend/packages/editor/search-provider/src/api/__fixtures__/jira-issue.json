{
  "scopes": [
    {
      "id": "jira.issue",
      "results": [
        {
          "id": "128578",
          "name": "FE error causing publishing errors - External Media Node must have a 'url' attribute",
          "url": "https://product-fabric.atlassian.net/browse/EDM-627",
          "attributes": {
            "@type": "issue",
            "key": "EDM-627",
            "issueTypeId": "10002",
            "issueTypeName": "Task",
            "containerId": "13055",
            "container": {
              "title": "Editor Media",
              "id": "13055"
            },
            "updated": "2020-07-08T14:59:56.112+1000",
            "avatar": {
              "url": "https://product-fabric.atlassian.net/secure/viewavatar?size=medium&avatarId=10318&avatarType=issuetype"
            }
          }
        },
        {
          "id": "162716",
          "name": "[BE] [SLA] [Edit Page] External media nodes are missing 'url' attribute",
          "url": "https://product-fabric.atlassian.net/browse/CERN-374",
          "attributes": {
            "@type": "issue",
            "key": "CERN-374",
            "issueTypeId": "11415",
            "issueTypeName": "Bug",
            "containerId": "12981",
            "container": {
              "title": "Confluence Editor - Reactor Core",
              "id": "12981"
            },
            "updated": "2020-07-08T00:43:47.608+1000",
            "avatar": {
              "url": "https://product-fabric.atlassian.net/secure/viewavatar?size=medium&avatarId=10303&avatarType=issuetype"
            }
          }
        },
        {
          "id": "167819",
          "name": "Media file fails to insert",
          "url": "https://product-fabric.atlassian.net/browse/EDM-833",
          "attributes": {
            "@type": "issue",
            "key": "EDM-833",
            "issueTypeId": "10002",
            "issueTypeName": "Story",
            "containerId": "13055",
            "container": {
              "title": "",
              "id": "13055"
            },
            "updated": "2020-07-08T09:02:18.334+1000",
            "avatar": {
              "url": "https://product-fabric.atlassian.net/secure/viewavatar?size=medium&avatarId=10318&avatarType=issuetype"
            }
          }
        },
        {
          "id": "169212",
          "name": "Create Editor Media amplitude dashboard",
          "url": "https://product-fabric.atlassian.net/browse/EDM-848",
          "attributes": {
            "@type": "issue",
            "key": "EDM-848",
            "issueTypeId": "10002",
            "issueTypeName": "Task",
            "containerId": "13055",
            "container": {
              "title": "Editor Media",
              "id": "13055"
            },
            "updated": "2020-07-08T09:02:18.548+1000",
            "avatar": {
              "url": "https://product-fabric.atlassian.net/secure/viewavatar?size=medium&avatarId=10318&avatarType=issuetype"
            }
          }
        },
        {
          "id": "170464",
          "name": "We've had a bunch of performance issues in Smart Links - all fixed, but we should monitor/instrument them (4 votes)",
          "url": "https://product-fabric.atlassian.net/browse/EDM-888",
          "attributes": {
            "@type": "issue",
            "key": "EDM-888",
            "issueTypeId": "10002",
            "issueTypeName": "",
            "containerId": "13055",
            "container": {
              "title": "Editor Media",
              "id": "13055"
            },
            "updated": "2020-06-19T16:00:12.589+1000",
            "avatar": {
              "url": "https://product-fabric.atlassian.net/secure/viewavatar?size=medium&avatarId=10318&avatarType=issuetype"
            }
          }
        }
      ],
      "size": 4654,
      "abTest": {
        "abTestId": "filters-experiment",
        "controlId": "filters-experiment",
        "experimentId": "filters-experiment"
      }
    }
  ]
}
