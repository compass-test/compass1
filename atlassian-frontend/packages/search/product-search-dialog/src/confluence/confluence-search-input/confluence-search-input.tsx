import React, { FunctionComponent } from 'react';
import {
  ProductSearchInput,
  BaseInputProps,
} from '../../common/product-search-input';
import { injectIntl, InjectedIntlProps } from 'react-intl';
import { messages } from '../../messages';
import { usePrimarySiteConfluenceAdvancedSearchUrlFactory } from '../confluence-utils/confluence-url-utils';
import { AdvancedSearchLinkSubjectId } from '../../common/analytics';

interface AdditionalProps {
  debounceTime: number;
  forwardRef: React.Ref<HTMLInputElement>;
  isLoading: boolean;
  onNavigate: (href: string, event: React.MouseEvent | KeyboardEvent) => void;
  onOpen: () => void;
}

export type Props = BaseInputProps & AdditionalProps;

const ConfluenceSearchInputBase: FunctionComponent<
  Props & InjectedIntlProps
> = ({ intl, ...rest }) => {
  const confluenceAdvancedSearchUrlFactory = usePrimarySiteConfluenceAdvancedSearchUrlFactory();
  const advancedSearchUrl = confluenceAdvancedSearchUrlFactory(rest.value);
  return (
    <ProductSearchInput
      {...rest}
      advancedSearchURL={advancedSearchUrl}
      actionSubjectId={AdvancedSearchLinkSubjectId.CONFLUENCE}
      expandedPlaceholder={intl.formatMessage(
        messages.confluence_search_input_expanded_placeholder,
      )}
      collapsedPlaceholder={intl.formatMessage(
        messages.common_search_input_collapsed_placeholder,
      )}
    />
  );
};

export const ConfluenceSearchInput = injectIntl(ConfluenceSearchInputBase);

export default ConfluenceSearchInput;
