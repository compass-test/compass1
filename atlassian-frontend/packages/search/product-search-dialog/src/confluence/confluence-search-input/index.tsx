import * as ConfluenceSearchInputImports from './confluence-search-input';

export { default as ConfluenceSearchInput } from './confluence-search-input';

export type Props = ConfluenceSearchInputImports.Props;
