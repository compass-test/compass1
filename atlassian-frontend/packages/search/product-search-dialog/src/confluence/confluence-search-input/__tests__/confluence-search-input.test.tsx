import React from 'react';
import { ConfluenceSearchInput } from '../confluence-search-input';
import { mountWithIntl } from '../../../__tests__/__fixtures__/intl-test-helpers';
jest.mock('../../confluence-utils/confluence-url-utils', () => ({
  usePrimarySiteConfluenceAdvancedSearchUrlFactory: () => (query: string) =>
    `some/advanced/search/url?q=${query}`,
}));

jest.mock('../../../common/product-search-input', () => {
  return { ProductSearchInput: (props: any) => <div {...props} /> };
});

const advancedSearchSelected = jest.fn();
const searchSessionId = 'testSessionId';
const isLoading = false;

const commonProps = {
  searchSessionId,
  advancedSearchSelected,
  isLoading,
  onOpen: () => {},
  debounceTime: 0,
  forwardRef: null,
  onNavigate: jest.fn(),
  value: '',
};

describe('<ConfluenceSearchInput />', () => {
  it('matches snapshot when expanded', () => {
    const wrapper = mountWithIntl(
      <ConfluenceSearchInput {...commonProps} isExpanded />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot when collapsed', () => {
    const wrapper = mountWithIntl(
      <ConfluenceSearchInput {...commonProps} isExpanded={false} />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
