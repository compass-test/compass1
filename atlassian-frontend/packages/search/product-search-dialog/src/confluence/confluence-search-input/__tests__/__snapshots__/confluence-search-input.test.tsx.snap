// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfluenceSearchInput /> matches snapshot when collapsed 1`] = `
<ConfluenceSearchInputBase
  advancedSearchSelected={[MockFunction]}
  debounceTime={0}
  forwardRef={null}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  isExpanded={false}
  isLoading={false}
  onNavigate={[MockFunction]}
  onOpen={[Function]}
  searchSessionId="testSessionId"
  value=""
>
  <ProductSearchInput
    actionSubjectId="confluenceAdvancedSearchLink"
    advancedSearchSelected={[MockFunction]}
    advancedSearchURL="some/advanced/search/url?q="
    collapsedPlaceholder="Search"
    debounceTime={0}
    expandedPlaceholder="Search Confluence"
    forwardRef={null}
    isExpanded={false}
    isLoading={false}
    onNavigate={[MockFunction]}
    onOpen={[Function]}
    searchSessionId="testSessionId"
    value=""
  >
    <div
      actionSubjectId="confluenceAdvancedSearchLink"
      advancedSearchSelected={[MockFunction]}
      advancedSearchURL="some/advanced/search/url?q="
      collapsedPlaceholder="Search"
      debounceTime={0}
      expandedPlaceholder="Search Confluence"
      forwardRef={null}
      isExpanded={false}
      isLoading={false}
      onNavigate={[MockFunction]}
      onOpen={[Function]}
      searchSessionId="testSessionId"
      value=""
    />
  </ProductSearchInput>
</ConfluenceSearchInputBase>
`;

exports[`<ConfluenceSearchInput /> matches snapshot when expanded 1`] = `
<ConfluenceSearchInputBase
  advancedSearchSelected={[MockFunction]}
  debounceTime={0}
  forwardRef={null}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  isExpanded={true}
  isLoading={false}
  onNavigate={[MockFunction]}
  onOpen={[Function]}
  searchSessionId="testSessionId"
  value=""
>
  <ProductSearchInput
    actionSubjectId="confluenceAdvancedSearchLink"
    advancedSearchSelected={[MockFunction]}
    advancedSearchURL="some/advanced/search/url?q="
    collapsedPlaceholder="Search"
    debounceTime={0}
    expandedPlaceholder="Search Confluence"
    forwardRef={null}
    isExpanded={true}
    isLoading={false}
    onNavigate={[MockFunction]}
    onOpen={[Function]}
    searchSessionId="testSessionId"
    value=""
  >
    <div
      actionSubjectId="confluenceAdvancedSearchLink"
      advancedSearchSelected={[MockFunction]}
      advancedSearchURL="some/advanced/search/url?q="
      collapsedPlaceholder="Search"
      debounceTime={0}
      expandedPlaceholder="Search Confluence"
      forwardRef={null}
      isExpanded={true}
      isLoading={false}
      onNavigate={[MockFunction]}
      onOpen={[Function]}
      searchSessionId="testSessionId"
      value=""
    />
  </ProductSearchInput>
</ConfluenceSearchInputBase>
`;
